### PR TITLE
New resource: azurerm_logic_app_integration_account_partner

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Currently, TF doesn't support Logic App Integration Account Partner. So submitted this PR to implement it.

--- PASS: TestAccLogicAppIntegrationAccountPartner_basic (277.01s)
--- PASS: TestAccLogicAppIntegrationAccountPartner_complete (279.23s)
--- PASS: TestAccLogicAppIntegrationAccountPartner_requiresImport (296.33s)
--- PASS: TestAccLogicAppIntegrationAccountPartner_update (389.14s)

API Reference:
https://github.com/Azure/azure-rest-api-specs/blob/ceb6d11a85d834c838181a3031cd106d14a2d674/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L4602

### CODE REVIEW

File: internal/services/logic/logic_app_integration_account_partner_resource.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"integration_account_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IntegrationAccountName(),
+			},
+
+			"business_identity": {
```
Comment (katbyte) [inline internal/services/logic/logic_app_integration_account_partner_resource.go:56]: this should be plural?
```suggestion
			"business_identities": {
```

File: website/docs/r/logic_app_integration_account_partner.html.markdown [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+* `business_identity` - (Optional) A `business_identity` block as documented below.
+
+* `metadata` - (Optional) A JSON mapping of any Metadata for this Logic App Integration Account Partner.
+
+---
+
+A `business_identity` block exports the following:
+
+* `qualifier` - (Required) The business identity qualifier for the Logic App Integration Account Partner.
```
Comment (katbyte) [inline website/docs/r/logic_app_integration_account_partner.html.markdown:58]: what is a business identity qualifier

File: website/docs/r/logic_app_integration_account_partner.html.markdown [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+* `value` - (Required) The user defined business identity value for the Logic App Integration Account Partner.
```
Comment (katbyte) [inline website/docs/r/logic_app_integration_account_partner.html.markdown:60]: what does this value represent/mean/what format/type

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/logic/logic_app_integration_account_partner_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/logic/logic_app_integration_account_partner_resource.go:56]: I assume here should be singular. I assume the field name should be plural when the situation is below this. If I am wrong, please correct me.

```
"match_values": {
	Type:     pluginsdk.TypeSet,
	Required: true,
	Elem: &pluginsdk.Schema{
		Type: pluginsdk.TypeString,
	},
},
```

Comment (katbyte) [reply internal/services/logic/logic_app_integration_account_partner_resource.go:56]: 👍

File: website/docs/r/logic_app_integration_account_partner.html.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/logic_app_integration_account_partner.html.markdown:58]: Updated. Found the description from https://docs.microsoft.com/en-us/azure/logic-apps/logic-apps-enterprise-integration-partners

File: website/docs/r/logic_app_integration_account_partner.html.markdown (hunk 3 above)
Comment (neil-yechenwei) [reply website/docs/r/logic_app_integration_account_partner.html.markdown:60]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , Thanks for your comments. I've updated code. Please have another look. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 🏗️

**Pull request merged**